### PR TITLE
(PUP-2400) Add mount acceptance tests

### PR DESCRIPTION
**Pull request by johnduarte**
Add basic acceptance tests for mount type. This adds acceptance
tests for defined, mounted, query, modify, and destroy.

### CODE REVIEW

**Comment by kurtwall** [comment]
Failing on Fedora 20 thus:
```
     51 fw8f9sy6cq8ivdv.delivery.puppetlabs.net (agent) 15:27:40$ umount /pl872469
     52 umount: /pl872469: mountpoint not found
     53
     54 fw8f9sy6cq8ivdv.delivery.puppetlabs.net (agent) executed in 0.03 seconds
     55 Exited: 32
     56 #<Beaker::Host::CommandFailure: Host 'fw8f9sy6cq8ivdv.delivery.puppetlabs.net'      56 exited with 32 running:
     57  umount /pl872469
     58 Last 10 lines of output were:
     59         umount: /pl872469: mountpoint not found>
```

**Comment by joshcooper** [comment]
@johnduarte I'm thinking we should target stable for inclusion in 4.1.0.

File: acceptance/tests/resource/mount/defined.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  #------- SETUP -------#
+  step "(setup) backup fstab file"
+  on(agent, "cp #{fstab} /tmp/fstab", :acceptable_exit_codes => [0,1])
+
+  #------- TESTS -------#
+  step "create a mount with puppet (defined)"
+  args = ['ensure=defined',
+          "device='/tmp/#{name}'"
```
Comment (kurtwall) [inline acceptance/tests/resource/mount/defined.rb]: I had to add `"fstype=>'ext4'"` (for example) to get this test to pass on Fedora and some other platforms. https://github.com/kurtwall/puppet/commit/f6bf93c31d16f58591743a514f52eddb1edd5655

Comment (johnduarte) [inline acceptance/tests/resource/mount/defined.rb]: @kurtwall  that sounds like a bug in the provider not the test.

@joshcooper shouldn't the provider supply sane defaults for the `fstype` based on the OS detail?

**Comment by kurtwall** [comment]
Passed in testing on RHEL7, F21, F20, Ubuntu 14.10, and Windows 2012r2.

**Pull request merged**